File: frontend/src/interfaces/certificate.ts
//import type { Employee } from "./employee";

export interface Certificatemanage {
    ID: number;
    title_th: string;
    title_en: string;
    detail: string;
    organizer: string;
    picture_url: string;
    signature_1_url: string;
    signature_2_url: string;
    date: string;
    type: string;

    //em_id?: number;
    //employee?: Employee;
}

export interface CreateCertificateRequest {
    ID: number;
    title_th: string;
    title_en: string;
    detail: string;
    organizer: string;
    picture_url: string;
    signature_1_url: string;
    signature_2_url: string;
    date: string;
    type: string;

    //   em_id?: number;
    //   employee?: Employee
}

export interface UpdateCertificateRequest {
    ID: number;
    title_th: string;
    title_en: string;
    detail: string;
    organizer: string;
    picture_url: string;
    signature_1_url: string;
    signature_2_url: string;
    date: string;
    type: string;
    //   em_id?: number;
    //   employee?: Employee;
}

export interface DeleteCertificateRequest {
    ID: number;
}

export interface GetCertificateResponse extends Certificatemanage { }

File: backend/entity/certificate.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Certificate struct {
	gorm.Model
	TitleTH       string    `gorm:"not null" json:"title_th"`
	TitleEN       string    `json:"title_en"`
	Detail        string    `json:"detail"`
	Organizer     string    `json:"organizer"`
	PictureURL    string    `json:"picture_url"`
	Signature1URL string    `json:"signature_1_url"`
	Signature2URL string    `json:"signature_2_url"`
	Date          time.Time `json:"date"`
	Type          string    `gorm:"not null" json:"type"`

	// Foreign keys
	UserID         uint   `gorm:"not null" json:"user_id"`
	ResultID       string `gorm:";not null" json:"result_id"`
	RegistrationID uint   `gorm:"not null" json:"registration_id"`
}
